_posts/2016-05-12-Developers-love-spaces.md

```diff
@@ -0,0 +1,54 @@
+title: Developers love spaces - Solita Developer Survey 2016 results
+author: arto
+excerpt: Now we know what developers love and love not this year
+categories: 
+- Java
+tags: 
+- Java
+- Clojure 
+- Javascript
+- Development
+- Culture
+- Solita
+---
+
+## What do devs love?
+
+Well a lot of things, apparently. Today, being a dev is about being passionate about things. But in this case, I think I already let out a spoiler in the title. Devs do love spaces. And, thanks to recent results from Solita Developer Survey, I got the stats to prove it. According the the survey, crushing percentage of 72,2% of all solita devs indent their code with spaces, as opposed to tabs. 2% didn't understand the question, so I suppose they don't indent.
+
+![Would you indent code with tabs or spaces?](/img/devs-love-spaces/spaces-vs-tabs.png)
+
+Other trends are also visible in the survey: for maintaining configurations - an important part of doing devops, or doing things right - there are many tool choices available. Here in Solita, the preferred choice is Ansible - with all its pros and cons 42% of devs choose it for automating and documenting the configurations. Chef and Puppet together don't come even close. Many still use traditional shell scripting (21%), or have to do manual work (16%). Another way to look at these statistics is to say that 84% of developers work in happily automated configuration management world, while the rest 16% just have to suck it.
+
+![How do you manage?](/img/devs-love-spaces/configuration-management.png)
+
+Then there're other, more obvious statistics. When it comes to version control, Git is way cooler than anything else currently. Most popular tools are IntelliJ IDEA, and Eclipse, while there're still solid 2,4% Vim-coders out there. For rapid text edit work people often whip out Sublime or Notepad++. Most devs are running OSX on their machines here, with almost equal amount running Windows. 12% are running some brand of Linux. 
+
+## How about frameworks and languages?
+
+Most popular languages used here are: Java, Clojure, Javascript, and C#. Anything else is just peanuts when it comes to statistics. Interestingly, Clojure is the most preferred language for devs, so I suppose we just can't get enough of Clojure in daily work.
+
+For the frontend development, most of the work is done with Angular, with React + Reagent catching up. Most people would choose to use React for the next project - but then most people haven't yet been exposed to Angular 2 so we'll see about that in 2017 ;) 
+
+## Roles and experience
+
+Most of our coders have been in business over 11 years - so they've seen a lot. 95% have been coding at least 6 years - or 75% if we count only coding for bread and butter, not for fun.
+
+It seems most of people in dev role consider themselves as fullstack developers - pretty much half - with 10% specialized back-end developers, 10% architects, and 5% slices of front-end developers, and integration specialists.
```

Comment:
most people in a dev role
...luulisin että virkkeen vika pilkku menee väärin, eli joko ilman tai
, and 5% of int specialists.